src/api.h

```diff
@@ -39,25 +34,297 @@ AWS_CRT_API const char *aws_crt_error_debug_str(int);
+
+/* X509 credentials provider */
+typedef struct _aws_crt_credentials_provider_x509_options aws_crt_credentials_provider_x509_options;
+AWS_CRT_API aws_crt_credentials_provider_x509_options *aws_crt_credentials_provider_x509_options_new(void);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_release(aws_crt_credentials_provider_x509_options *options);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_set_thing_name(
+    aws_crt_credentials_provider_x509_options *options,
+    const uint8_t *thing_name,
+    size_t thing_name_length);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_set_role_alias(
+    aws_crt_credentials_provider_x509_options *options,
+    const uint8_t *role_alias,
+    size_t role_alias_length);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_set_endpoint(
+    aws_crt_credentials_provider_x509_options *options,
+    const uint8_t *endpoint,
+    size_t endpoint_length);
+AWS_CRT_API aws_crt_credentials_provider *aws_crt_credentials_provider_x509_new(
+    aws_crt_credentials_provider_x509_options *options);
+
+/* STS Web Identity provider */
+typedef struct aws_credentials_provider_sts_web_identity_options aws_crt_credentials_provider_sts_web_identity_options;
+AWS_CRT_API aws_crt_credentials_provider_sts_web_identity_options *
+    aws_crt_credentials_provider_sts_web_identity_options_new(void);
+AWS_CRT_API void aws_crt_credentials_provider_sts_web_identity_options_release(
+    aws_crt_credentials_provider_sts_web_identity_options *options);
+AWS_CRT_API aws_crt_credentials_provider *aws_crt_credentials_provider_sts_web_identity_new(
+    const aws_crt_credentials_provider_sts_web_identity_options *options);
+
+/* aws_signing_config_aws */
+typedef struct _aws_crt_signing_config_aws aws_crt_signing_config_aws;
+typedef enum aws_crt_signing_algorithm {
+    AWS_CRT_SIGNING_ALGORITHM_V4 = 0,
+} aws_crt_signing_algorithm;
+typedef enum aws_crt_signature_type {
+    AWS_CRT_SIGNATURE_TYPE_HTTP_REQUEST_HEADERS,
+    AWS_CRT_SIGNATURE_TYPE_HTTP_REQUEST_QUERY_PARAMS,
+    AWS_CRT_SIGNATURE_TYPE_HTTP_REQUEST_CHUNK,
+    AWS_CRT_SIGNATURE_TYPE_HTTP_REQUEST_EVENT,
+    AWS_CRT_SIGNATURE_TYPE_CANONICAL_REQUEST_HEADERS,
+    AWS_CRT_SIGNATURE_TYPE_CANONICAL_REQUEST_QUERY_PARAMS,
+} aws_crt_signature_type;
+typedef enum aws_crt_signed_body_header_type {
+    AWS_CRT_SIGNED_BODY_HEADER_TYPE_NONE,
+    AWS_CRT_SIGNED_BODY_HEADER_TYPE_X_AMZ_CONTENT_SHA256,
+} aws_crt_signed_body_header_type;
+AWS_CRT_API aws_crt_signing_config_aws *aws_crt_signing_config_aws_new(void);
+AWS_CRT_API void aws_crt_signing_config_aws_release(aws_crt_signing_config_aws *signing_config);
+
+AWS_CRT_API void aws_crt_signing_config_aws_set_algorithm(
+    aws_crt_signing_config_aws *signing_config,
+    enum aws_crt_signing_algorithm algorithm);
+AWS_CRT_API void aws_crt_signing_config_aws_set_signature_type(
+    aws_crt_signing_config_aws *signing_config,
+    enum aws_crt_signature_type sig_type);
```

Comment:
trivial: you typedefed these 2 enum types, so don't need to write `enum` as part of the argument type